src/environmentd/src/http/sql.rs

```diff
@@ -53,11 +56,75 @@ struct ErrorResponse {
     error: String,
 }
 
-pub async fn handle_sql_ws(client: AuthedClient, ws: WebSocketUpgrade) -> impl IntoResponse {
-    ws.on_upgrade(|ws| async move { run_ws(client, ws).await })
+pub async fn handle_sql_ws(
+    State(state): State<WsState>,
+    ws: WebSocketUpgrade,
+) -> impl IntoResponse {
+    ws.on_upgrade(|ws| async move { run_ws(&state, ws).await })
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+//#[serde(untagged)]
+pub struct WebSocketAuth {
+    pub user_id: String,
+    pub password: String,
+}
+
+async fn init_ws(
+    WsState {
+        frontegg,
+        adapter_client,
+    }: &WsState,
+    ws: &mut WebSocket,
+) -> Result<AuthedClient, anyhow::Error> {
```

Comment:
nit: feel like this should go in the http.rs module, next to the other auth code, just to keep all the auth stuff in one place.